File: UPTMDigital.API/Models/Asistencia.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace UPTMDigital.API.Models
{
    [Table("Asistencia")]
    public class Asistencia
    {
        [Key]
        public int IdAsistencia { get; set; }
        public int AsignaturaId { get; set; }
        public int EstudianteId { get; set; }
        public DateTime Fecha { get; set; }
        public string Estado { get; set; } = "Presente";
        public string? CodigoQR { get; set; }
        public int? ProfesorId { get; set; }

        public Asignatura Asignatura { get; set; } = null!;
        public Estudiante Estudiante { get; set; } = null!;
        public Profesor? Profesor { get; set; }
    }
}

File: uptmdigital_app/lib/models/asistencia.dart
class Asistencia {
  final int idAsistencia;
  final int asignaturaId;
  final int estudianteId;
  final String fecha;
  final String codigoQR;

  Asistencia({
    required this.idAsistencia,
    required this.asignaturaId,
    required this.estudianteId,
    required this.fecha,
    required this.codigoQR,
  });

  factory Asistencia.fromJson(Map<String, dynamic> json) {
    return Asistencia(
      idAsistencia: json['idAsistencia'] ?? 0,
      asignaturaId: json['asignaturaId'] ?? 0,
      estudianteId: json['estudianteId'] ?? 0,
      fecha: json['fecha'] != null ? json['fecha'].toString() : '',
      codigoQR: json['codigoQR'] ?? '',
    );
  }
}
